src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3971,7 +3971,7 @@ private string ScanNumberHelper(char firstChar, out NumberFormat format, out Num
                         case 'X':
                             sb.Append('0'); // Prepend a 0 to the number before any numeric digits are added
                             ScanHexDigits(sb);
-                            if (sb.Length == 0)
+                            if (sb.Length == 1) // 1 instead of 0 to account for append ↑↑
```

Comment:
Looking at this again... I think there's another case this will still fall into; `-0x.ps1` will still trip up this code path, I think.

I think we'll need:

1. Set a `hasSign` bool variable at the start of this ScanNumberHelper method to `false`
2. In the `if (firstChar.IsDash() || firstChar == '+')` branch we need to set `hasSign = true`.
3. This check can be `if ((hasSign && sb.Length == 2) || sb.Length == 1)`

Reply:
E.g.
`-0x.ps1` > null output (as expected currently)
`Powershell -0x.ps1` > The numeric constant -0x is not valid. `BadNumericConstant`

In fact even when writing a test case for this scenario I'm having to search for workarounds that will let me run the script. 
![image](https://user-images.githubusercontent.com/19718175/197363082-9abf30e3-a37f-44ff-b7a5-945aace95483.png)

Reply:
Same goes for `+0x.ps1` obviously.